src/main/java/at/hannibal2/skyhanni/features/combat/ArmorStackDisplay.kt

```diff
@@ -33,10 +49,39 @@ object ArmorStackDisplay {
         display = stacks
     }
 
+    @SubscribeEvent
+    fun onSecond(event: SecondPassedEvent) {
+        if (config.showTimer) {
+            InventoryUtils.getArmor().forEach { armor ->
+                armorStackTierBonus.findMatcher(armor?.getLore().toString()) {
+                    group("amount")
+                }?.let { result ->
+                    armorMaxTime = when (result) {
+                        "2" -> 4000.milliseconds
+                        "3" -> 7000.milliseconds
+                        "4" -> 10000.milliseconds
+                        else -> ZERO
+                    }
+                    return
+                }
+            }
+        }
+    }
+
+    @SubscribeEvent
+    fun onActionBarValueUpdate(event: ActionBarValueUpdateEvent) {
+        if (event.updated != ActionBarStatsData.ARMOR_STACK) return
+        displayTimer = SimpleTimeMark.now().plus(armorMaxTime)
+    }
+
     @SubscribeEvent
     fun onRenderOverlay(event: GuiRenderEvent.GuiOverlayRenderEvent) {
         if (!isEnabled()) return
         config.position.renderString(display, posLabel = "Armor Stack Display")
+
+        if (config.showTimer && displayTimer.isInFuture()) {
+            config.timerPosition.renderString(displayTimer.timeUntil().toString(DurationUnit.SECONDS, 2), posLabel = "Armor Stack Timer Display")
```

Comment:
Ideally, the display should only be calculated in `SecondPassedEvent`, stored in a string, then be displayed in this event.

Reply:
Doesn't that make decimals update once a second?

Reply:
true, but you should use the .format function we have for duration already instead of this

Reply:
good points. Additionally please merge this into one display, and update it only 10 times a second (every second tick), instead of every frame.